src/c++/library/http_client.cc

```diff
@@ -609,6 +727,12 @@ class InferResultHttp : public InferResult {
   InferResultHttp(std::shared_ptr<HttpInferRequest> infer_request);
   InferResultHttp(const Error err) : status_(err) {}
 
+  ~InferResultHttp();
+
+  Error ConvertJSONOutputToBinary(
```

Comment:
Yuck. We have new classes defined in the .cc file?  @debermudez it would be nice to clean this up and split into multiple files.

Reply:
Indeed. A lot of the c++ client could use some refactoring.